app/src/main/java/org/projectbuendia/client/ui/chart/ObsFormat.java

```diff
@@ -456,7 +456,7 @@ private boolean matches(@Nullable Object obj, String operator, String operandStr
                 case "":
                 case "=":
                 case "==":
-                    return Objects.equal(value, operand);
+                    return Objects.equal(value.uuid, operand.uuid);
```

Comment:
Oops, it looks like this line 459 change got committed.  Would you mind reverting it?